File: src/com/company/LinkList.java
package com.company;

import java.io.IOException;

public class LinkList {
    private Link first;

    public LinkList() {
        this.first = null;
    }

    public boolean isEmpty(){
        return (first == null);
    }

    public void insertFirst (int id, double dd)
    {
        Link newLink = new Link(id, dd);
        newLink.next = first;
        first = newLink;
    }

    public Link deleteFirst()
    {
        Link temp = first;
        first = first.next;
        return temp;
    }

    public Link findKey (int key)
    {
        Link currentLink = first;
        if (currentLink == null)
            return null;
        else
        {
            while (currentLink.iData != key)
            {
                if (currentLink.next == null)
                    return null;
                else
                {
                    currentLink = currentLink.next;
                }
            }
            return currentLink;
        }
    }

    public Link deleteKey (int key)
    {
        Link previousLink = first;
        Link currentLink = first;
        if (previousLink == null)
            return null;
        else
        {
            while (currentLink.iData != key)
            {
                if (currentLink.next == null)
                    return null;
                else
                {
                    previousLink = currentLink;
                    currentLink = currentLink.next;
                }
            }
            if (currentLink == first)
            {
                first = currentLink.next;
            }
            previousLink.next = currentLink.next;
            return currentLink;
        }
    }

    public void displayList(){
       try
       {
           System.out.print("Link: ");
           Link current = first;
           while (current.next != null)
           {
               current.displayLink();
               current = current.next;
           }
           System.out.println(" :Link");
       } catch (Exception e)
       {
           System.out.println("Я Русский!");
       }

    }
}

File: StructuresOnKotlin/src/LinkList.kt
class LinkList(private var first:Link? = null) {

    fun insertFirst(i:Int, d:Double)
    {
      val newLink = Link(i,d)
      newLink.next = first
        first = newLink
    }

    fun deleteFirst(): Link?
    {
        val temp:Link? =  first
        first = first?.next
        return temp
    }

    fun displayList()
    {
        var current:Link? = first
        while (current !=null)
        {
            current.display()
            current = current.next
        }
        println()
    }

    fun displayThis(){
        first?.display()
        println("/////")
        first?.next?.display()
    }

    fun findByKey(i:Int):Link?
    {
        val current:Link? = first
        while (current?.getiData() != i)
        {
            if (current?.next == null)
            {
                println("findByKey совпадений не найдено")
                return current
            }
            current.next = current
        }
        return current
    }

    fun deleteByKey(i:Int):Link?
    {
        val current:Link? = first
        var previous:Link? = first
        while (current?.getiData() != i)
        {
            if (current?.next == null)
            {
                println("deleteByKey совпадений не найдено")
                return current
            }
            previous = current
            current.next = current
        }
        return if (current == first)
        {
            first = first?.next
            current
        }
        else
        {
            previous?.next= current.next
            current
        }
    }

    fun isEmpty() = first == null


}